cmd/machine-api-operator/start.go

```diff
@@ -101,3 +115,42 @@ func startControllers(ctx *ControllerContext) {
 		recorder,
 	).Run(2, ctx.Stop)
 }
+
+func startMetricsCollectionAndServer(ctx *ControllerContext) {
+	machineInformer := ctx.MachineInformerFactory.Machine().V1beta1().Machines()
+	machinesetInformer := ctx.MachineInformerFactory.Machine().V1beta1().MachineSets()
+	machineMetricsCollector := metrics.NewMachineCollector(
+		machineInformer,
+		machinesetInformer,
+		componentNamespace)
+	go func() {
+		if !cache.WaitForCacheSync(ctx.Stop,
+			machineInformer.Informer().HasSynced,
+			machinesetInformer.Informer().HasSynced) {
+			glog.Fatal("Failed to sync caches")
+		}
+		glog.Info("Synced up machine api informer caches")
+	}()
+	prometheus.MustRegister(machineMetricsCollector)
```

Comment:
should this be inside the go subroutine so the WaitForCacheSync blocks the rest of the execution?